src/oops/SOLID/DependencyInversionPrinciple/before/payments/PaymentProcessor.java

```diff
@@ -1,30 +1,19 @@
 package oops.SOLID.DependencyInversionPrinciple.before.payments;
 
-import java.util.List;
-
 import oops.SOLID.DependencyInversionPrinciple.before.employees.Employee;
-import oops.SOLID.DependencyInversionPrinciple.before.notification.EmailSender;
-import oops.SOLID.DependencyInversionPrinciple.before.repositories.EmployeeFileRepository;
-import oops.SOLID.DependencyInversionPrinciple.before.serielizer.EmployeeFileSerializer;
+import oops.SOLID.DependencyInversionPrinciple.before.repositories.EmployeeRepository;
 
 public class PaymentProcessor {
 
+    private final EmployeeRepository employeeRepository;
 
-    public int sendPayments(){
-    	EmployeeFileSerializer serializer =
-                new EmployeeFileSerializer();
-    	
-    	EmployeeFileRepository employeeRepository = 
-    			new EmployeeFileRepository(serializer);
-    	
-        List<Employee> employees = employeeRepository.findAll();
-        int totalPayments = 0;
-
-        for(Employee employee : employees){
-            totalPayments += employee.getMonthlyIncome();
-            EmailSender.notify(employee);
-        }
+    public PaymentProcessor(final EmployeeRepository employeeRepository){
+        this.employeeRepository = employeeRepository;
+    }
 
-        return totalPayments;
+    public int sendPayments(){
```

Comment:
You aren't sending the email notification as part of this implementation.

Reply:
Yeah right! I forgot it